### PR TITLE
Marketplace: Theme Installation page redirecting to the Thank You Page

### LINKED ISSUE #2045: Notifications: spamming a comment should not auto hide that comment
Raised by @gibrown 

I was reading my notifications and I accidentally hit a key which i guess spammed the comment. While trying to parse the "comment spammed" ui and figure out how to undo it, it timed out, and disappeared. I have no idea which p2 it was on. I have no idea which post. So now I am stuck trying to hunt down which comment it was.

Eventually I found the comment, but it required going to individual p2s, into wp-admin, and guessing where it might have been. Given that we are relying on notifications to "manage" comments, it was a really awful user experience.

Note from @drw158 

> Couple of ideas:
> - For destructive actions, keep the alert around longer, or possibly forever until the user hits the X to close it.
> - Remove keyboard shortcuts for destructive actions, or require a more complex shortcut (Shift + S).

### PR DESCRIPTION
**Pull request by WBerredo**
Fixes [Marketplace: Transfer to Atomic when activating a Theme#2045](https://github.com/Automattic/dotcom-forge/issues/2045)

## Proposed Changes
* Add an installation route for themes
* Update texts and illustrations for themes
* Add theme installation via direct URL access
* Add atomic transformation for themes

## Testing Instructions

* From an atomic site
* Go to the theme installation route with a dotorg slug. Ex: `/marketplace/theme/astra/install/:site`
* Click to activate the theme
* wait for the theme activation and you should be redirected to the thank you page with the theme selected

---

* From a simple site
* Go to the theme installation route with a dotorg slug. Ex: `/marketplace/theme/astra/install/:site`
* Click to activate the theme
* wait for the theme activation and atomic and you should be redirected to the thank you page with the theme selected, similar to the previous step

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/state/data-layer/wpcom/sites/atomic/transfers/index.js [hunk 1]
```diff
@@ -8,17 +8,41 @@ import { registerHandlers } from 'calypso/state/data-layer/handler-registry';
...
+};
+
 const initiateAtomicTransfer = ( action ) =>
 	http(
 		{
 			apiNamespace: 'wpcom/v2',
 			method: 'POST',
 			path: `/sites/${ action.siteId }/atomic/transfers/`,
-			...( action.softwareSet
+			...( action.softwareSet || action.themeSlug
```
Comment (gcsecsey) [inline client/state/data-layer/wpcom/sites/atomic/transfers/index.js:43]: Shouldn't we allow `action.pluginSlug` here as well?

**Review by gcsecsey — APPROVED** [review]
Thanks @WBerredo for tackling this, it needed a fair amount of modification to work! 🙌 🎉 

Tests are passing for me and the installation works as described for both atomic and non-atomic sites! 🚢 

#### Simple site
Installation:
CleanShot 2023-04-05 at 12 37 50@2x

After clicking Activate:
CleanShot 2023-04-05 at 12 38 25@2x

#### Atomic site
CleanShot 2023-04-05 at 12 38 25@2x

Note: First I couldn't test the changes on Calypso live, visiting `/marketplace/theme/astra/:site` redirected to `/home/:site` for all my sites. I've checked the code again and saw that the install URL has an extra `/install` in it, so I've updated the test steps with this.

**Comment by WBerredo** [comment]
Thanks for updating the steps @gcsecsey, the `/marketplace/theme/astra/install/:site` is the route for themes.

File: client/state/data-layer/wpcom/sites/atomic/transfers/index.js (hunk 1 above)
Comment (WBerredo) [reply client/state/data-layer/wpcom/sites/atomic/transfers/index.js:43]: We use a different method for atomic transformation with plugins, so the plugin slug is not needed here.

https://github.com/Automattic/wp-calypso/blob/50f6cee2b3db04d0c300fc4ff970225fade2f2c5/client/my-sites/marketplace/pages/marketplace-plugin-install/index.tsx#L215-L220

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7911430

Hi @WBerredo, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `Do you want to activate the theme %(theme)s?`
- `Go to the theme page`
- `Activate theme`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7911430).

**Pull request merged**